cli/src/main/java/hudson/cli/FullDuplexHttpStream.java

```diff
@@ -49,15 +49,36 @@ private static String basicAuth(String userInfo) {
     }
 
     /**
+     * @param target something like {@code http://jenkins/cli?remoting=true}
+     *               which we then need to split into {@code http://jenkins/} + {@code cli?remoting=true}
+     *               in order to construct a crumb issuer request
+     * @deprecated use {@link #FullDuplexHttpStream(URL, String, String)} instead
+     */
+    @Deprecated
+    public FullDuplexHttpStream(URL target, String authorization) throws IOException {
+        this(new URL(target.toString().replaceFirst("/cli.*$", "/")), target.toString().replaceFirst("^.+/(cli.*)$", "$1"), authorization);
```

Comment:
The only usage of this constructor was from `CLI`, which passed for example `http://jenkins/cli`. I am performing the same transformation that the code previously performed in the crumb retrieval.